models/intermediate/int_sap__0employee_date_changes.sql

```diff
@@ -0,0 +1,108 @@
+{% set employee_models = ['pa0000', 'pa0001', 'pa0007', 'pa0008', 'pa0031'] %}
+
+with
+{% for employee_model in employee_models %}
+    {{ employee_model }}_beg as (
+        select
+            pernr,
+            begda as date_change
+        from {{ var(employee_model) }}
+    ),
+
+    {{ employee_model }}_end as (
+        select
+            pernr,
+			case when endda = '99991231' then endda
+				else
+				--convert date strings to dates, add one day, and back to avoid null records as we would join on date ranges in the end model that don't exist
+				{% if target.type in ('redshift', 'snowflake', 'snowflake-sap') %}
+				to_char(dateadd(day, 1, to_date(endda, 'YYYYMMDD')), 'YYYYMMDD') end as date_change
+				{% elif target.type == 'postgres' %}
+				to_char(to_date(endda, 'YYYYMMDD') + interval '1 day', 'YYYYMMDD') end as date_change
+				{% elif target.type == 'bigquery' %}
+				format_date('%Y%m%d', date_add(parse_date('%Y%m%d', endda), interval 1 day)) end as date_change
+				{% elif target.type == 'databricks' %}
+				date_format(date_add(to_date(endda, 'yyyyMMdd'), 1), 'yyyyMMdd') end as date_change
+				{% endif %}
+		from {{ var(employee_model) }}
+	), 
+{% endfor %}
+
+unioned_dates as (
+
+	{% for employee_model in employee_models %}
+	
+		{% if target.type == 'bigquery' %}
```

Comment:
Union Distinct is unique to BigQuery. It is the equivalent of Union in all other warehouses: https://blog.coupler.io/bigquery-union-syntax-and-usage-examples/#BigQuery_UNION_DISTINCT

I do not believe union distinct in other warehouses (and union doesn't exist in bigquery), although admittedly I was in a hurry to finish this part since I only discovered it late in my Buildkite runs during the sprint,.

Reply:
Is there a reason we can't just use `union all`?

Reply:
`union all` led to duplicate records if there were multiple changes on the same date from the different models funneling into the employee date changes model. So we'd end up with pernrs with the same dates.

But I see your point that it's a lot of repetitive code, so I applied a union all, then added an additional group by cte.